docs/docs/creating-dynamically-rendered-navigation.md

```diff
@@ -2,7 +2,176 @@
 title: Creating Dynamically-Rendered Navigation
 ---
 
-This is a stub. Help our community expand it.
 
-Please use the [Gatsby Style Guide](/docs/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+## Creating dynamic navigation in Gatsby
+
+Often at times you will want to be able to easily edit your websites navigation in response to a *change in requirements*. To achieve this, you can use gatsby to dynamically generate your navigation. Where you store the data for your navigation can be anywhere a backend API, CMS, headless CMS or even the filesystem.
+
+What this doc will cover:
+
+ - Adding data to your siteMetaData
+ - Querying it using graphql
+ - Pulling the data into a component using StaticQuery
+ - Dynamically generating a navigation component
+
+You will be using `gatsby-config.js` to store the data for your links. `gatsby-config.js` is a file used for configuring gatsby located in the root path of every Gatsby project. A plain old javascript object is exported from this file, this object contains `siteMetadata` object which you can query through graphql when generating your static pages. 
+
+This guide will use the gatsby starter project  `gatsby-starter-default`, which can be downloaded through the gatsby CLI.  
+
+### Creating the link data
+First, locate the `gatsby-config.js` file in the root directory of your project. Inside the `siteMetaData` object, add an array of menu links objects. These objects should contain two properties: name and link. Name is the name of your navigation item, link is the page which will be navigated to when a menu item is clicked.
+```js
+module.exports = {
+  siteMetadata: {
+    title: 'Gatsby Default Starter',
+    menuLinks:[
+      {name:'home',link:'/'},
+      {name:'page2',link:'/page-2'}
+    ]
+  },
+  plugins: []
+}
+```
+### Viewing the `siteMetaData` in graphql
+
+You can use graphql to query for information container in  `siteMetadata` object located in your projects `gatsby-config.js`. Inside your project directory run `npm run develop` to start a development server running or port 8000.
+
+Navigate to `http://localhost:8000/___graphql` to view the GraphiQL editor created by facebook that enables you to quickly test graphql queries against datasources. Using the documentation explorer you can view the current GraphQL schema for your project which is an invaluable resource during development.
+
+Examining the available types in GraphQL you will come across a site query that looks as follows:
+
+```js
+site(
+    siteMetadata: siteSiteMetadataInputObject_2
+    port: sitePortQueryString_2
+    host: siteHostQueryString_2
+    pathPrefix: sitePathPrefixQueryString_2
+    polyfill: sitePolyfillQueryBoolean_2
+    buildTime: siteBuildTimeQueryString_2
+    id: siteIdQueryString_2
+    internal: siteInternalInputObject_2
+): Site
```

Comment:
I don't think that is necessary here - this could be useful in graphql related docs, but it feels like getting side tracked into graphql details